c7n/resources/s3.py

```diff
@@ -1344,12 +1344,17 @@ class SetBucketReplicationConfig(BucketActionBase):
     def process(self, buckets):
         with self.executor_factory(max_workers=3) as w:
             futures = {w.submit(self.process_bucket, bucket): bucket for bucket in buckets}
+            errors = list()
             for future in as_completed(futures):
                 bucket = futures[future]
                 try:
                     future.result()
                 except Exception as e:
-                    self.log.error('Message: %s Bucket: %s', e, bucket['Name'])
+                    err = 'Message: %s Bucket: %s', e, bucket['Name']
+                    self.log.error(err)
+                    errors.append(err)
+            if errors:
+                raise Exception(errors)        
```

Comment:
Raise errors instead of just log them

Reply:
Exception typically takes a string, passing a list of errors, is a little different, I think you want to `raise Exception('\n'.join(map(str, errors)))`

Reply:
:) that output looks much better! Thanks for the review!! I just sent an updated commit.